source/validate.cpp

```diff
@@ -261,6 +267,143 @@ spv_result_t spvValidateIDs(const spv_instruction_t* pInsts,
   return SPV_SUCCESS;
 }
 
+// TODO(umar): Validate header
```

Comment:
A subtle thing:  the Id bound should be validated also.  But you can only do that after you've seen all the instructions in the module.